presto-main/src/test/java/io/prestosql/operator/scalar/time/TestExtract.java

```diff
@@ -156,4 +157,284 @@ public void testMillisecond()
         assertThat(assertions.expression("millisecond(TIME '12:34:56.12345678901')")).matches("BIGINT '123'");
         assertThat(assertions.expression("millisecond(TIME '12:34:56.123456789012')")).matches("BIGINT '123'");
     }
+
+    @Test
+    public void testYear()
```

Comment:
ideally all TestExtract classes should have same order of methods, so we can easily find what is and what is not implemented yet.

`io.prestosql.operator.scalar.timestamptz.TestExtract` is the most comprehensive today, and it lays out the method in from longest fields to shortest (year first, then to millis)

Reply:
Except that for `date` that's overkill. Each test is a one-liner, so it's easier to read if they are all in the same method.

Reply:
As a sidenote, we may want to deprecate fields other than year, month, day, hour, minute, second, timezone_hour, timezone_minute and rely on the existing dedicated functions for others. The spec defines EXTRACT as:

> `<extract expression>` operates on a datetime or interval and returns an exact numeric value representing the value of one **component** of the datetime or interval.

None of the other "fields" we support are really components, but transformations.